src/OpenTelemetry.Contrib.Instrumentation.Quartz/Implementation/QuartzDiagnosticListener.cs

```diff
@@ -0,0 +1,139 @@
+﻿// <copyright file="QuartzDiagnosticListener.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using OpenTelemetry.Instrumentation;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Contrib.Instrumentation.Quartz.Implementation
+{
+    internal sealed class QuartzDiagnosticListener : ListenerHandler
+    {
+        internal static readonly AssemblyName AssemblyName = typeof(QuartzDiagnosticListener).Assembly.GetName();
+        internal static readonly string ActivitySourceName = AssemblyName.Name;
+        internal static readonly Version Version = AssemblyName.Version;
+        internal static readonly ActivitySource ActivitySource = new ActivitySource(ActivitySourceName, Version.ToString());
+
+        private readonly QuartzInstrumentationOptions options;
+
+        public QuartzDiagnosticListener(string sourceName, QuartzInstrumentationOptions options)
+            : base(sourceName)
+        {
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            if (activity.IsAllDataRequested)
+            {
+                if (!this.options.TracedOperations.Contains(activity.OperationName))
+                {
+                    QuartzInstrumentationEventSource.Log.OperationIsFilteredOut(activity.OperationName);
+                    activity.IsAllDataRequested = false;
+                    return;
+                }
+
+                activity.DisplayName = this.GetDisplayName(activity);
+
+                ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ActivitySource);
+                ActivityInstrumentationHelper.SetKindProperty(activity, this.GetActivityKind(activity));
+
+                try
+                {
+                    this.options.Enrich?.Invoke(activity, "OnStartActivity", payload);
```

Comment:
similar instrumentations we have (http, aspnetcore), extracts the required objects from the payload using the property fetchers, so end user always get the right type, and don't have to deal with the payload extraction themselves. Any reason why we don't that approach here? 
Same comment for exception as well..

Reply:
@CodeBlanch to share your thoughts on this approach, where the payload is passed as-is to the Enrich, vs extracting the strong type object (JobDetail etc), and passing it to Enrich.